crates/extensions/c8y_mapper_ext/src/service_monitor.rs

```diff
@@ -17,6 +26,28 @@ fn default_status() -> String {
     "unknown".to_string()
 }
 
+impl HealthStatus {
+    fn from_mosquitto_bridge_payload_str(payload: &str) -> Self {
+        let status = match payload {
+            MQTT_BRIDGE_UP_PAYLOAD => UP_STATUS,
+            MQTT_BRIDGE_DOWN_PAYLOAD => DOWN_STATUS,
+            _ => UNKNOWN_STATUS,
+        };
```

Comment:
This logic should be moved into `tedge_api`. It would even be better to have a function taking both the topic and the payload. Indeed, with the current API (a bunch of constants), this is up to the caller to figure out that this is a bridge health topic and that the payload has to be decoded differently).

Reply:
Yes, a comprehensive cleanup of the currently fragmented health monitoring code will be done a follow-up PR as mentioned in this [comment](https://github.com/thin-edge/thin-edge.io/pull/2903#issuecomment-2133431005)